src/modules/create_institution/app/create_institution_usecase.ts

```diff
@@ -0,0 +1,58 @@
+import { randomUUID } from 'crypto';
+import { Institution } from "../../../core/structure/entities/Institution";
+import { MissingParameter } from '../../../core/helpers/errors/ModuleError';
+import { IInstitutionRepo } from '../../../core/repositories/interfaces/IInstitutionRepo';
+import { IUserRepo } from '../../../core/repositories/interfaces/IUserRepo';
+
+export class CreateInstitutionUsecase {
+    public database_repo: IInstitutionRepo;
+    public user_repo: IUserRepo;
+
+    constructor(database_repo: IInstitutionRepo, user_repo: IUserRepo) {
+        this.database_repo = database_repo;
+        this.user_repo = user_repo;
+    }
+
+    public async execute(institution: Institution, headers: any){
+        if (!headers.authorization) {
+            throw new MissingParameter("Authorization");
+        }
+        
+        const userExists = await this.user_repo.get_user(headers.authorization);
+        if (!userExists) {
+            throw new Error("User does not exist in our database");
+        }
+
```

Comment:
After getting the user, check if his type is ADMIN or MODERATOR, else throw UserNotAllowed